packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanel.tsx

```diff
@@ -0,0 +1,188 @@
+
+        let value = e.target.value;
+        let newModel;
+
+        if (value == 0) {
+            newModel = model.merge({
+                keyPropertyId: undefined,
+                isDemographicData: true,
+                keyPropertyManaged: false
+            }) as DatasetModel;
+        }
+        else if (value == 1) {
+            newModel = model.merge({
+                keyPropertyId: undefined,
+                isDemographicData: false,
+                keyPropertyManaged: false
+            }) as DatasetModel;
+        }
+        else {
+            newModel = model.merge({
+                keyPropertyId: 0, // resetting key property id
+                isDemographicData: false
+            }) as DatasetModel;
+        }
+        this.updateValidStatus(newModel);
+    };
+
+    onAdditionalKeyFieldChange = (name, formValue, selected): void => {
+        this.onChange(name, selected.propertyId);
+    };
+
+
+    applyAdvancedProperties = (advancedSettingsForm: DatasetAdvancedSettingsForm) => {
+        const { model } = this.props;
+        const newModel = model.merge(advancedSettingsForm) as DatasetModel;
+        this.updateValidStatus(newModel);
+    };
+
+
+    render() {
+        const {
+            model,
+            showDataspace,
+        } = this.props;
+
+        const {
+            isValid
+        } = this.state;
+
+        return (
+            <BasePropertiesPanel
+                {...this.props}
+                headerId={'dataset-header-id'}
+                title={'Dataset Properties'}
+                titlePrefix={model.name}
+                isValid={isValid}
+                updateValidStatus={this.updateValidStatus}
+            >
+                <Row className={'margin-bottom'}>
+                    <Col xs={12}>
+                        <HelpTopicURL helpTopic={DEFINE_DATASET_TOPIC} nounPlural={'datasets'}/>
+                    </Col>
+                </Row>
+                <Form>
+
+                    <Col xs={12} md={5}>
+                        <BasicPropertiesFields
+                            model={model}
+                            onInputChange={this.onInputChange}
+                            onCategoryChange={this.onCategoryChange}
+                        />
+                    </Col>
+
+                    <Col xs={12} md={5}>
+                        <DataRowUniquenessContainer
+                            model={model}
+                            onRadioChange={this.onDataRowRadioChange}
+                            onCheckBoxChange={this.onInputChange}
+                            onSelectChange={this.onAdditionalKeyFieldChange}
+                        />
+                    </Col>
+
+                    <AdvancedSettings
+                        title={"Advanced Settings"}
+                        model={model}
+                        showDataspace={showDataspace}
+                        applyAdvancedProperties={this.applyAdvancedProperties}
+                    />
```

Comment:
wrap this in `<Col>` to make the total equal 12 for md:
```suggestion
<Col xs={12} md={2}>
                    <AdvancedSettings
                        title={"Advanced Settings"}
                        model={model}
                        showDataspace={showDataspace}
                        applyAdvancedProperties={this.applyAdvancedProperties}
                    />
</Col>
```

Reply:
I tried this but doing this makes the advanced settings button off from the right side.

Reply:
Oh, might need to add the "pull-right" className.

Reply:
does not seem to work
![Screen Shot 2020-04-03 at 2 09 29 PM](https://user-images.githubusercontent.com/39965420/78405200-d9641780-75b4-11ea-9255-d9819ef9f2f9.png)

Reply:
Ah ok, I didn't notice this before, but looks like those usages of `<Col>` are not in a `<Row>`, adding that in locally did the trick (and I didn't need the pull-right) part. I also don't think we need the `<Form>` element here.